desktop/cmp/grouping/GroupingChooser.js

```diff
@@ -0,0 +1,399 @@
+        emptyText = 'Ungrouped',
+        popoverWidth = 250,
+        popoverTitle = 'Group By',
+        popoverPosition = 'bottom'
+    }) {
+        const {editorIsOpen, favoritesIsOpen, value} = model,
+            isOpen = editorIsOpen || favoritesIsOpen,
+            getButtonText = () => {
+                const prefix = buttonValueTextPrefix,
+                    dimText = model.getValueLabel(value);
+                return prefix ? `${prefix} ${dimText}` : dimText;
+            },
+            getButtonTitle = () => {
+                const labels = model.getValueLabel(value).split(' > ');
+                return labels.map((it, i) => ' '.repeat(i) + (i ? '› ' : '') + it).join('\n');
+            };
+
+        return div({
+            className,
+            item: popover({
+                isOpen,
+                popoverRef: model.popoverRef,
+                popoverClassName: classNames('xh-grouping-chooser-popover', editorIsOpen ? 'xh-popup--framed' : null),
+                position: favoritesIsOpen ? 'bottom-right' : popoverPosition,
+                minimal: favoritesIsOpen,
+                target: fragment(
+                    button({
+                        item: buttonText ?? (isEmpty(model.value) ? emptyText : getButtonText()),
+                        title: buttonTitle ?? (isEmpty(model.value) ? emptyText : getButtonTitle()),
+                        icon: buttonIcon,
+                        width: buttonWidth,
+                        className: classNames('xh-grouping-chooser-button', styleButtonAsInput ? 'xh-grouping-chooser-button--as-input' : null),
+                        minimal: styleButtonAsInput,
+                        onClick: () => model.showEditor()
+                    }),
+                    favoritesIcon()
+                ),
+                content: favoritesIsOpen ? favoritesMenu() : editor({popoverWidth, popoverTitle}),
+                onInteraction: (nextOpenState, e) => {
+                    if (nextOpenState === false) {
+                        // Prevent clicks with Select controls from closing popover
+                        const selectPortal = document.getElementById(Select.MENU_PORTAL_ID)?.contains(e?.target),
+                            selectClick = e?.target?.classList.contains('xh-select__single-value');
+                        if (!selectPortal && !selectClick) model.closePopover();
+                    }
+                }
+            })
+        });
+    }
+});
+
+GroupingChooser.propTypes = {
+    /** Icon for target button. */
+    buttonIcon: PT.element,
+
+    /** Static text for target button, or null (default) to display current dimensions. */
+    buttonText: PT.node,
+
+    /**
+     * Prefix for button text - applied when value not-empty and static text not specified.
+     *      E.g. "Group by" to render "Group by Fund > Trader".
+     */
+    buttonValueTextPrefix: PT.node,
+
+    /** Width in pixels of the target button. */
+    buttonWidth: PT.number,
```

Comment:
Here I would also vote for simplifying / standardizing to `width` - we can keep the distinct `popoverWidth` prop so this (by process of elimination) would pretty clearly apply to the button.  

⭐  Would also like to ensure we support setting a `flex` prop